src/main/java/ch/njol/skript/classes/data/DefaultFunctions.java

```diff
@@ -711,6 +705,227 @@ public String[] executeSimple(Object[][] params) {
+			.examples(
+				"root(2, 4) = 2 # same as sqrt(4)",
+				"root(4, 16) = 2",
+				"root(-4, 16) = 0.5 # same as 16^(-1/4)"
+			)
+			.since("INSERT VERSION");
+
+		Functions.registerFunction(new SimpleJavaFunction<Number>("permutation", new Parameter[]{
+			new Parameter<>("options", DefaultClasses.NUMBER, true, null),
+			new Parameter<>("size", DefaultClasses.NUMBER, true, null)
+		}, DefaultClasses.NUMBER, true) {
+			@Override
+			public Number @Nullable [] executeSimple(Object[][] params) {
+				Double options = ((Number) params[0][0]).doubleValue();
+				Double size = ((Number) params[1][0]).doubleValue();
+				if (size > options || size < 0) {
+					return null;
+				} else if (size.equals(0d)) {
+					return new Number[]{1};
+				} else if (size.equals(1d)) {
+					return new Number[]{options};
+				}
+				Double result = 1d;
+				for (double i = options; i > options - size; i--) {
+					if (result.isInfinite() || result.isNaN())
+						break;
+					result *= i;
+				}
+				return new Number[]{result};
+			}
+		})
+			.description(
+				"Get the number of possible ordered arrangements from 1 to 'options' with each arrangement having a size equal to 'size'",
+				"For example, a permutation with 3 options and an arrangement size of 1, returns 3: (1), (2), (3)",
+				"A permutation with 3 options and an arrangement size of 2 returns 6: (1, 2), (1, 3), (2, 1), (2, 3), (3, 1), (3, 2)"
+			)
+			.examples(
+				"permutation(10, 2) = 90",
+				"permutation(10, 4) = 5040",
+				"permutation(size of {some list::*}, 2)"
+			)
+			.since("INSERT VERSION");
+
+		Functions.registerFunction(new SimpleJavaFunction<Number>("combination", new Parameter[]{
+				new Parameter<>("options", DefaultClasses.NUMBER, true, null),
+				new Parameter<>("size", DefaultClasses.NUMBER, true, null)
+			}, DefaultClasses.NUMBER, true) {
+				@Override
+				public Number @Nullable [] executeSimple(Object[][] params) {
+					Double options = ((Number) params[0][0]).doubleValue();
+					Double size = ((Number) params[1][0]).doubleValue();
+					if (size > options || size < 0) {
+						return null;
+					} else if (size.equals(0d)) {
+						return new Number[]{1};
+					} else if (size.equals(1d)) {
+						return new Number[]{options};
+					}
+					Double top = 1d;
+					for (double i = options; i > options - size; i--) {
+						if (top.isInfinite() || top.isNaN())
+							return new Number[]{top};
+						top *= i;
+					}
+					Double bottom = size;
+					for (double i = size - 1; i > 1; i--) {
+						if (bottom.isInfinite() || bottom.isNaN())
+							break;
+						bottom *= i;
+					}
+					return new Number[]{top/bottom};
+				}
+			})
+			.description(
+				"Get the number of possible sets from 1 to 'options' with each set having a size equal to 'size'",
+				"For example, a combination with 3 options and a set size of 1, returns 3: (1), (2), (3)",
+				"A combination of 3 options with a set size of 2 returns 3: (1, 2), (1, 3), (2, 3)"
```

Comment:
These two functions should also have a disclaimer about approximations too, like factorial.